GPU/Workflow/src/GPUWorkflowSpec.cxx

```diff
@@ -410,51 +411,19 @@ DataProcessorSpec getGPURecoWorkflowSpec(gpuworkflow::CompletionPolicyData* poli
       }
       if (specconfig.zsDecoder) {
         std::vector<InputSpec> filter = {{"check", ConcreteDataTypeMatcher{gDataOriginTPC, "RAWDATA"}, Lifetime::Timeframe}};
-        for (auto const& ref : InputRecordWalker(pc.inputs(), filter)) {
-          const DataHeader* dh = DataRefUtils::getHeader<DataHeader*>(ref);
-          if (dh->payloadSize == 0 && dh->subSpecification == 0xDEADBEEF) {
-            LOG(INFO) << "Received 0xDEADBEEF message with no RAW input, performing dummy processing on emtpry input data";
-            continue; // Dummy message inserted if there is no input, just ignore
-          }
-          const gsl::span<const char> raw = pc.inputs().get<gsl::span<char>>(ref);
-          o2::framework::RawParser parser(raw.data(), raw.size());
-
-          const unsigned char* ptr = nullptr;
-          int count = 0;
-          rdh_utils::FEEIDType lastFEE = -1;
-          int rawcru = 0;
-          int rawendpoint = 0;
-          size_t totalSize = 0;
-          for (auto it = parser.begin(); it != parser.end(); it++) {
-            const unsigned char* current = it.raw();
-            const RAWDataHeader* rdh = (const RAWDataHeader*)current;
-            if (current == nullptr || it.size() == 0 || (current - ptr) % TPCZSHDR::TPC_ZS_PAGE_SIZE || o2::raw::RDHUtils::getFEEID(*rdh) != lastFEE) {
-              if (count) {
-                tpcZSmetaPointers[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(ptr);
-                tpcZSmetaSizes[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(count);
-              }
-              count = 0;
-              lastFEE = o2::raw::RDHUtils::getFEEID(*rdh);
-              rawcru = o2::raw::RDHUtils::getCRUID(*rdh);
```

Comment:
@davidrohr, you mean, that these lines have been outdated for a while already? How did it work then? I will change the updated code to use `rdh_utils::get*` as you suggested

Reply:
We don't have real TPC data in the final ZS format, and in the MC data the rawcru in the RDH is set to the logical TPC CRU, so there it was working (more or less by chance).

Reply:
ok